### PR TITLE
Appsv1 pilot

### PR DESCRIPTION
**Pull request by irisdingbj**
check in changes for pilot to consume K8s changes :
These resources have been deprecated since k8s v1.9 and are planned to no longer be served starting in k8s v1.16 (c.f. kubernetes/kubernetes#43214)

details are in #12211

### CODE REVIEW

File: pilot/pkg/kube/inject/webhook_test.go [hunk 1]
```diff
@@ -605,14 +605,12 @@ func TestWebhookInject(t *testing.T) {
 			wantFile:     "TestWebhookInject_http_probe_nosidecar_rewrite.patch",
 			templateFile: "TestWebhookInject_http_probe_nosidecar_rewrite_template.yaml",
 		},
-		{
-			inputFile:    "TestWebhookInject_https_probe_rewrite.yaml",
-			wantFile:     "TestWebhookInject_https_probe_rewrite.patch",
-			templateFile: "TestWebhookInject_https_probe_rewrite_template.yaml",
```
Comment (howardjohn) [inline pilot/pkg/kube/inject/webhook_test.go]: why was this removed?

Comment (irisdingbj) [reply pilot/pkg/kube/inject/webhook_test.go]: Typo.. fixed. Thanks!

File: pilot/pkg/kube/inject/webhook_test.go [hunk 2]
```diff
@@ -613,6 +613,9 @@ func TestWebhookInject(t *testing.T) {
 	}
 
 	for i, c := range cases {
+		if c.inputFile != "TestWebhookInject_http_probe_nosidecar_rewrite.yaml" {
```
Comment (howardjohn) [inline pilot/pkg/kube/inject/webhook_test.go]: Why is this here?

Comment (irisdingbj) [reply pilot/pkg/kube/inject/webhook_test.go]: Another typo.. :(

**Comment by irisdingbj** [comment]
@linsun @rshriram Need your review here to get this in. Thanks!

**Comment by irisdingbj** [comment]
@linsun @frankbu 

samples need be updated to point to apps/v1 as well.

For Doc update, I think we should update to point the selector req as well.

**Comment by frankbu** [comment]
@linsun, @irisdingbj The bookinfo sample has already been updated, but the other samples are still TBD. I don't think any Istio doc needs to be updated - this has nothing to do with Istio, it's a standard Kubernetes requirement.

**Comment by linsun** [comment]
@frankbu I was referring to this section - is this still required if user has fulfilled the k8s requirement for the selector blob?

```
Deployments with app and version labels: We recommend adding an explicit app label and version label to deployments. Add the labels to the deployment specification of pods deployed using the Kubernetes Deployment. The app and version labels add contextual information to the metrics and telemetry Istio collects.
The app label: Each deployment specification should have a distinct app label with a meaningful value. The app label is used to add contextual information in distributed tracing.
The version label: This label indicates the version of the application corresponding to the particular deployment.
```

**Comment by howardjohn** [comment]
The app and version label are used for telemetry/tracing stuff. Selector is used by Kubernetes. I don't think we need to say anything about the selector, it is irrelevant to Istio afaik

**Comment by linsun** [comment]
@howardjohn @douglas-reid Is it possible for telemetry to leverage the selector used by k8s vs asking our users to do additional things just for istio?

**Comment by howardjohn** [comment]
I think the selector is completely unrelated to Istio. The selector just links a deployment to pods. https://stackoverflow.com/a/50309744 has a great explanation.

**Comment by linsun** [comment]
i'm not saying it is related to istio, i'm asking if we could leverage this existing requirement from k8s from istio telemetry vs creating the deployment label requirements onto our users like our doc has described.

**Comment by frankbu** [comment]
@linsun, well, the selector labels and deployment labels are really used for two different things, so trying to use one for both purposes seems kind of kludgy. As @howardjohn said, the purpose of the selector labels is to identify the pods associated with the deployment. They are a new K8s requirement. The deployment labels are used for selecting deployments. I'm assuming that the telemetry/tracing want them for that purpose, so I think we need them all. In general, they don't always need to be the exact same labels, although it sounds like the selector ones need to be the same (subset?) of the pod labels.

**Comment by linsun** [comment]
@frankbu i know they are different purpose... i'm mainly trying to reduce our pods requirements for service mesh to 0 as user often forgets to read that and it is very hard to troubleshoot issues when something is left out.
